src/status_im2/contexts/chat/lightbox/zoomable_image/view.cljs

```diff
@@ -0,0 +1,313 @@
+(ns status-im2.contexts.chat.lightbox.zoomable-image.view
+  (:require
+   [react-native.core :as rn]
+   [react-native.gesture :as gesture]
+   [react-native.reanimated :as reanimated]
+   [reagent.core :as reagent]
+   [utils.re-frame :as rf]))
+
+;; Definitions
+(def min-scale 1)
+
+(def double-tap-scale 2)
+
+(def max-scale 5)
+
+(def init-offset 0)
+
+(def velocity-factor 0.25)
+
+(def default-duration 300)
+
+;; Some aliases for reanimated methods, as they are used 10s of times in this file
+(defn get-val
+  [animation]
+  (reanimated/get-shared-value animation))
+
+(defn set-val
+  [animation value]
+  (reanimated/set-shared-value animation value))
+
+(defn use-val
+  [value]
+  (reanimated/use-shared-value value))
```

Comment:
I don't understand the purpose of these wrapper functions, they just create indirection and pass along the unmodified arguments. I think we should just stick to the normal `use-shared-value`, `set-shared-value`, etc.

I guess you really wanted shorter names :wink:, but I think if that's the case we should consider renaming the original functions in a separate PR if it bothers other people too, but we should avoid needless wrapper functions.

Reply:
These are some aliases for reanimated methods (for shorter names), as they are used 10s of times in this file. Using `reanimated/set-shared-value` causes long lines all over the file.

I don't know what can we rename it to. It seems to me it is more of a clojure problem due to namespacing and kebab-case. Using it in JS would be `setSharedValue` vs in clojure `reanimated/set-shared-value` (14 vs 27 chars, almost double :(  ) 

When it occuers few times in a file, it is not a problem, but for this particular file, it looks a bit unreadable and ugly.

Reply:
I understand where you're coming from, but I disagree @OmarBasem. These wrapper functions only add complication to an already complicated code (pretty much all heavily animated code is difficult to digest). There's a better way.

>These are some aliases for reanimated methods (for shorter names), as they are used 10s of times in this file. Using reanimated/set-shared-value causes long lines all over the file.

The team decided to use long and explicit namespace aliases. And there's nothing stopping us from renaming `set-shared-value` to `set-val` in `react-native.reanimated`. This change alone would make the code much shorter, but let's not stop there.

Much more importantly, **the problem is not at the language (Clojure) level, it's at the abstraction level**. The repeated calls to `reanimated/set-shared-value` or `set-val` are ugly and unnecessary. Here's a better idea IMO, change the function to accept setting multiple values at once, so that the name of the function becomes way less relevant.

Instead of this:

```clojure
(set-val pan-x (timing init-offset duration))
(set-val pinch-x (timing init-offset duration))
(set-val pan-x-start (timing init-offset duration))
(set-val pinch-x-start (timing init-offset duration))
(set-val pan-y (timing init-offset duration))
(set-val pinch-y (timing init-offset duration))
(set-val pinch-y-start (timing init-offset duration))
(set-val pan-y-start (timing init-offset duration))
```

Better! Notice I used the word `animated` instead of `reanimated` too. This is common practice in Javascript projects.

```clojure
(animated/set-val
 pan-x (timing init-offset duration)
 pinch-x (timing init-offset duration)
 pan-x-start (timing init-offset duration)
 pinch-x-start (timing init-offset duration)
 pan-y (timing init-offset duration)
 pinch-y (timing init-offset duration)
 pinch-y-start (timing init-offset duration)
 pan-y-start (timing init-offset duration))
```

Reply:
For creating true aliases in Clojure, you should use the pattern:

```clojure
(def set-val reanimated/set-shared-value)
```

But as I commented, I don't like this solution at all. Much better is to rename `reanimated/set-shared-value` to `reanimated/set-val` or something and also make the function accept multiple values.

Reply:
But hey @OmarBasem, I'm not suggesting you implement any of the changes above in this PR. I'm just saying we should do the right thing that moves the needle for everybody, and creating aliases just for this namespace is a dubious practice at best.

Reply:
Thanks @ilmotta. You are right, we need to abstract it in a better way. I will try to make a PR for that.

Reply:
I would gladly review it, thank you @OmarBasem

Reply:
Hi, Sorry for bumping this thread, but I just noticed that we are making abstraction changes in PR https://github.com/status-im/status-mobile/pull/15175 based on above comment.

> Better! Notice I used the word animated instead of reanimated too. This is common practice in Javascript projects.

[Animated](https://reactnative.dev/docs/animated) and [Reanimated](https://docs.swmansion.com/react-native-reanimated/) are different libraries and [Animated values](https://reactnative.dev/docs/animatedvalue) and [Shared values](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/shared-values/) are different things, abstracting and shortening names here will probably be misleading.

And as I remember, we are trying to avoid abstraction if possible to avoid confusion. Maybe @J-Son89 has some thoughts

Reply:
@Parveshdhull I am aware of these differences. But the current abstraction of reanimated is causing very long names, especially when used multiple times in a file. Using `reanimated/set-shared-value animation (reanimated/with-timing value)` 10s of times in a file makes it more difficult to read. 

That issue https://github.com/status-im/status-mobile/issues/15176 is an attempt to improve the abstraction. For example (just suggestions) by using animated instead of reanimated (which is what is actually used in reanimated docs), and using shorter names like `set-val` instead of `set-shared-value`.

Reply:
And probably we should get rid of React Native's Animated namespace/code since we shouldn't be using it and to avoid confusion.

Reply:
`reanimated/set-shared-value animation (reanimated/with-timing value)`
1. Here function name is only `set-shared-value` (16 chars) others are just params. If we are also counting them in name, then we are in trouble ([quo/button](https://github.com/status-im/status-mobile/blob/69c3a7235709c108621143f01edcd3a7754d5041/src/status_im2/contexts/syncing/sheets/sync_device_notice/view.cljs#L52))
2. You can just import functions with names and use like
```
[react-native.reanimated :refer [use-shared-value set-shared-value]]
(set-shared-value 4) 
```
  Only 21 chars, and you can get rid of reanimated
3. We do have a helper function for animating value with timing `animate-shared-value-with-timing`
 (It still has a bigger name :))
4. Changing `set-shared-value` to `set-value` only decreases your code from 68 chars to 61, not sure if it's worth the confusion it might create.
5. Even if the code is a little lengthier, in my opinion, all the parts are required to convey the work/process that the code is doing.

If similar code keeps repeating in your file, I guess might be ok to create localized abstraction(still not recommend) in that file, but not sure we need do this globally (as mentioned in https://github.com/status-im/status-mobile/issues/15176)

Reply:
I generally agree with this @Parveshdhull. I am okay with using `refer` for this case, and also okay with creating localized abstraction when required, but do not know what others think about that. If others are okay with that I can close the issue https://github.com/status-im/status-mobile/issues/15176.

cc @ilmotta @flexsurfer what are your thoughts guys?

Reply:
About the names:

When I read something as `reanimated/set-value` I know it's setting a shared value because of the namespace. The `shared` word is quite redundant in the context of reanimated. I also see that `reanimated` (using the prefix `re`) is redundant. In most React projects I've seen the alias is just `animated`. The vast majority of the time we're not mixing animated + reanimated code, so just `animated` as alias seems better IMO. Btw the alias name can be reinforced by the linter.

>And as I remember, we are trying to avoid abstraction if possible to avoid confusion.

@Parveshdhull I don't see any problem in re-evaluating the current abstractions and improving them as we go. It's a function just as any other function, nothing special about it being related to animations. Omar felt they're not very ergonomic, and I agree while reviewing his PRs.

>Only 21 chars, and you will get rid of reanimated

This *convenience argument* (less chars) is used for pretty much all usages of `:refer`, but lots of Clojure projects avoid them, because it creates a burden on the developer, that now needs to know where each function comes from to reason about the code. I've seen many well known Clojure devs recommending against `:refer` (I don't have the references obviously). I think our guidelines are err'ing on the side of readability, not shortness. We can have exceptions sure (we do have some in test namespaces), but each exception puts more burden on collaborators to know when to refer vs when to alias.

I'd say, let's keep avoiding `refer` and not open more exceptions. I can see a lot of repetition in other namespaces, but they're not using `:refer`, so why should we open an exception here?

Reply:
@OmarBasem ^^

Reply:
Hi @ilmotta,
Thank you very much for your valuable input.

If `use-val`, `set-val`, and `animated` are all understable then sounds awesome, I am all in for these changes. I just wanted to be sure we are not compromising the understandability for the short names.

> The shared word is quite redundant in the context of reanimated.

It will not be redundant in the case of an `animated` alias. Unless we check what we are importing.

> In most React projects I've seen the alias is just animated.

Just for curiosity, please can you refer to them?

> (omar): Using it in JS would be setSharedValue vs in clojure reanimated/set-shared-value (14 vs 27 chars, almost double :( )

I don't recommend refer too, it was a suggestion because of this statement.  That it's not a shortcoming of Clojure itself, we can achieve similar here.

@OmarBasem 
- If we are going to use these names for function, please make changes in the original file instead of creating another wrapper.
- `apply-animations-to-style` also sounds confusing I think we can rename this too into `use-animated-style` wdyt?
- Please can we also do these changes in separate PR. As https://github.com/status-im/status-mobile/pull/15175 is already big

Reply:
> It will not be redundant in the case of an animated alias. Unless we check what we are importing.

Good point @Parveshdhull

> In most React projects I've seen the alias is just animated.

I'm mostly referring to blog posts and courses, pretty much all use `Animated` to import the default module. Also the `reanimated` docs all use the `Animated` alias, so I guess that's one strong reason why the JavaScript community went for `Animated` instead of `Reanimated`.

It's nothing major, but it'd be nice if we could use the same alias, although I wouldn't like to confuse it with the actual animated module from RN. Well, perhaps better keep it as is to avoid confusion ;)

> Please can we also do these changes in separate PR. As https://github.com/status-im/status-mobile/pull/15175 is already big

Totally agree @Parveshdhull, it should be done in a separate PR.

Reply:
Thanks @ilmotta for your input!

Sure @Parveshdhull in a separate PR!